src/rpdk/core/contract/resource_client.py

```diff
@@ -22,7 +22,7 @@
     get_temporary_credentials,
 )
 from ..jsonutils.pointer import fragment_decode, fragment_list
-from ..jsonutils.utils import traverse
+from ..jsonutils.utils import item_hash, traverse
```

Comment:
Can we use Counter instead of generating our own hash?

Reply:
counter transforms into dictionary but in case collection has complex types (dictionaries) then we wont be able to hash then, that is why using md5 to custom hash dictionaries